share/functions/fish_svn_prompt.fish

```diff
@@ -116,11 +116,12 @@ function fish_svn_prompt --description "Prompt function for svn"
         # get the output for a particular column
         # 1. echo the whole status flag text
         # 2. cut out the current column of characters
-        # 3. remove spaces and newline characters
-        set -l column_status (printf '%s\n' $svn_status_lines | cut -c $col | sort -u | tr -d ' \n')
+        # 3. remove duplicates
+        # 4. remove spaces
+        set -l column_status (printf '%s\n' $svn_status_lines | cut -c $col | sort -u | tr -d ' ')
 
-        # check that the character count is not zero (this would indicate that there are status flags in this column)
-        if [ (count $column_status) -ne 0 ]
+        # check that the column status list does not only contain an empty element (if it does, this column is empty)
+        if [ (count $column_status) -gt 1 -o -n $column_status[1] ]
```

Comment:
`test -n` *always* needs to have the variable quoted.

Our `test` (or `[`) bultin follows POSIX, and that allows usage like `test string`, which does the same thing as `test -n string`. Unfortunately it does that by saying that `test` with any one argument needs to *always* be true, so `test -n`, `test -d` etc are all true.

(yes, this is awkward, sorry)

Reply:
Odd -- because it works ;-)
But just to be sure, I'll add quotes.